firmware/version_data.py

```diff
@@ -0,0 +1,117 @@
+import  subprocess
+import  os
+import  tempfile
+import  filecmp
+import  shutil
+
+commit          = subprocess.Popen(['git', 'log', '--format="%H"', '-n''1'], stdout=subprocess.PIPE)
```

Comment:
You should probably be using `subprocess.check_call` or `subprocess.check_output` calls instead of `subprocess.Popen` here.